modal/_utils/git_utils.py

```diff
@@ -0,0 +1,75 @@
+# Copyright Modal Labs 2025
+import os
+import subprocess
+from typing import Dict, List, Optional, Union
+
+
+def run_command(args: List[str]) -> Optional[str]:
+    """Run a command and return its output.
+
+    Args:
+        args: Command and arguments as a list
+
+    Returns:
+        Command output as string or None if command failed
+    """
+    try:
+        result = subprocess.run(
+            args,
+            check=True,
+            capture_output=True,
+            text=True,
+            shell=False,
+        )
+        return result.stdout.strip()
+    except (subprocess.CalledProcessError, FileNotFoundError):
+        return None
+
+
+def get_git_commit_info() -> Dict[str, Union[str, bool, int]]:
+    """Collect git information about the current repository."""
```

Comment:
I'm a little worried about the number of subprocess calls here adding nontrivial amount of latency to development loops. Feels like we could reduce the number of distinct calls out to `git` here?